client/cmd/dexcctl/main.go

```diff
@@ -41,6 +43,62 @@ func main() {
 	}
 }
 
+// promptPasswords is a map of routes to password prompts. Passwords are
+// prompted in the order given.
+var promptPasswords = map[string][]string{
+	"openwallet": {"App password:"},
+	"newwallet":  {"App password:", "Wallet password:"},
+	"init":       {"Set new app password:"},
+	"register":   {"App password:"},
+}
+
+// swapCerts is a map of routes to tsl certificate indexes.
+var swapCerts = map[string]int{
+	"preregister": 1,
+	"register":    2,
+}
+
+// promptPWs prompts for passwords on stdin and returns an error if prompting
+// fails or a password is empty. Returns passwords as a slice of strings.
+func promptPWs(cmd string) ([]string, error) {
+	prompts, exists := promptPasswords[cmd]
+	if !exists {
+		return nil, nil
+	}
+	pws := make([]string, len(prompts))
+	// Prompt for passwords one at a time.
+	for i, prompt := range prompts {
+		pw, err := admin.PasswordPrompt(prompt)
+		if err != nil {
+			return nil, err
+		}
+		pws[i] = string(pw)
+		admin.ClearBytes(pw)
+	}
+	return pws, nil
+}
+
+// swapPathsForFiles will find tsl certificate paths for routes in swapCerts,
+// attempt to read those files, and swap the path for the entire certificate as
+// a string.
+func swapPathsForFiles(cmd string, args *[]string) error {
+	idx, exists := swapCerts[cmd]
+	// Certs are optional, so it is not an error if they are not included.
+	if !exists || len(*args) < idx+1 || (*args)[idx] == "" {
+		return nil
+	}
+	path := cleanAndExpandPath((*args)[idx])
+	if !fileExists(path) {
+		return fmt.Errorf("no cert file found at %s", path)
+	}
+	certB, err := ioutil.ReadFile(path)
+	if err != nil {
+		fmt.Errorf("error reading %s: %v", path, err)
```

Comment:
Missing return.  Could use a test of `promptPWs` and `swapPathsForFiles` since they're a little complex.

Reply:
I tried a few things but I don't think I can make a test for proptPWs without changing the code a little bit. You can't just print to stdin, apparently. One suggested workaround https://stackoverflow.com/questions/38573176/mocking-crypto-ssh-terminal/54183219#54183219